File: step20_step_functions/step06_invoke_step_function_with_event/lambda/addData.ts
const { DynamoDB } = require("aws-sdk");

exports.handler = async (event: any) => {
  const dynamo = new DynamoDB();

  var generateId = Date.now();
  var idString = generateId.toString();

  const params = {
    TableName: process.env.DynamoTable,
    Item: {
      id: { S: idString },
      message: { S: event.detail.event },
    },
  };
  try {
    await dynamo.putItem(params).promise();
    return { operationSuccessful: true };
  } catch (err) {
    console.log("DynamoDB error: ", err);
    return { operationSuccessful: false };
  }
};

File: step20_step_functions/Python/step02_choice_part1/lambda/addData.py
from __future__ import print_function

import os
import boto3
import datetime


# Get the service resource.
dynamodb = boto3.resource('dynamodb')

# environment variable
TABLE_NAME = os.environ['DynamoTable']


def handler(event, context):
    table = dynamodb.Table(TABLE_NAME)

    generateId = datetime.datetime.now()
    idString = generateId.strftime("%m/%d/%Y, %H:%M:%S")    
    # put item in table    
    try:
        table.put_item(
        Item={
            'id': idString,
            'entry': 'New Entry Added'
            }
        )
        print('event', event)    
        print("PutItem succeeded:")
        return {'operationSuccessful':True}        
    except:        
        return {'operationSuccessful':False}
    
    
        
